subsys/net/lib/sockets/sockets.c

```diff
@@ -1134,6 +1134,75 @@ static int sock_get_pkt_src_addr(struct net_pkt *pkt,
 	return ret;
 }
 
+#if defined(CONFIG_NET_OFFLOAD)
+static bool net_pkt_remote_addr_is_unspecified(struct net_pkt *pkt)
+{
+	bool ret = true;
+
+	if (net_pkt_family(pkt) == AF_INET) {
+		ret = net_ipv4_is_addr_unspecified((struct in_addr *)
+								  &pkt->remote);
+	} else if (net_pkt_family(pkt) == AF_INET6) {
+		ret = net_ipv6_is_addr_unspecified((struct in6_addr *)
+								  &pkt->remote);
+	}
```

Comment:
This is not correct, you are casting `in_addr` and `in6_addr` to `sockaddr` which are not compatible. The address field is not the first field in the sockaddr struct so this will lead to incorrect value. I suggest this instead:
```
	if (net_pkt_family(pkt) == AF_INET) {
		ret = net_ipv4_is_addr_unspecified(&net_sin(&pkt->remote)->sin_addr);
	} else if (net_pkt_family(pkt) == AF_INET6) {
		ret = net_ipv6_is_addr_unspecified(&net_sin6(&pkt->remote)->sin6_addr);
	}
```

Reply:
Changed to suggested.